### PR TITLE
run eslint (with react config) on workers-playground/wrangler

### PR DESCRIPTION
**Pull request by threepointone**
This enables eslint (with our react config) for the workers-playground project. Additionally, this enables the react-jsx condition in relevant tsconfig/eslint configs, letting us write jsx without having React in scope.

## What this PR solves / how to test

Fixes https://github.com/cloudflare/workers-sdk/issues/6074

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: No functional changes
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: already covered 
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: No public changes

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -13237,7 +13237,7 @@ snapshots:
       eslint: 8.57.0
       eslint-import-resolver-node: 0.3.7
       eslint-import-resolver-typescript: 3.5.5(@typescript-eslint/parser@5.62.0(eslint@8.57.0)(typescript@5.4.5))(eslint-import-resolver-node@0.3.7)(eslint-plugin-import@2.27.5(@typescript-eslint/parser@5.62.0(eslint@8.57.0)(typescript@5.4.5))(eslint@8.57.0))(eslint@8.57.0)
-      eslint-plugin-import: 2.27.5(@typescript-eslint/parser@5.62.0(eslint@8.57.0)(typescript@5.4.5))(eslint-import-resolver-typescript@3.5.5)(eslint@8.57.0)
```
Comment (threepointone) [inline pnpm-lock.yaml:13240]: these changes happen when running `pnpm i` on main right now

File: packages/eslint-config-worker/react.js [hunk 2]
```diff
@@ -4,8 +4,12 @@ module.exports = {
 	plugins: ["eslint-plugin-react", "eslint-plugin-react-hooks"],
 	overrides: [
 		{
-			files: ["*.ts", "*.tsx"],
-			extends: ["plugin:react/recommended", "plugin:react-hooks/recommended"],
+			files: ["**/*.ts", "**/*.tsx"],
```
Comment (petebacondarwin) [inline packages/eslint-config-worker/react.js:7]: Do we want to check `*.ts` for JSX?

File: packages/workers-playground/src/QuickEditor/VSCodeEditor.tsx [hunk 3]
```diff
@@ -90,7 +90,7 @@ export function VSCodeEditor({ content, onChange }: Props) {
 	}, []);
 
 	useEffect(() => {
-		if (quickEdit !== null)
+		if (quickEdit !== null) {
```
Comment (petebacondarwin) [inline packages/workers-playground/src/QuickEditor/VSCodeEditor.tsx:93]: ❤️

File: packages/eslint-config-worker/react.js (hunk 2 above)
Comment (threepointone) [reply packages/eslint-config-worker/react.js:7]: there might be hooks implemented in ts files, that can still be linted

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**